src/main/java/com/aws/greengrass/authorization/AuthorizationModule.java

```diff
@@ -73,14 +80,13 @@ public boolean isPresent(final String destination, Permission permission) throws
         if (resource != null && Utils.isEmpty(resource)) {
             throw new AuthorizationException("Resource cannot be empty");
         }
-        if (!permissions.containsKey(destination)) {
-            return false;
-        }
-        List<Permission> permissionsForDest = permissions.get(destination);
-        if (!Utils.isEmpty(permissionsForDest)) {
-            return permissionsForDest.contains(permission);
+        Map<String, Map<String, WildcardVariableTrie>> destMap = amazingMap.get(destination);
+        if (destMap != null) {
```

Comment:
null isn't ever possible given the usage of DefaultConcurrentHashMap

Reply:
This is to check if there was no mapping for this key. it was using `containsKey()` but that results in one extra nested 'if' and PMD complains

Reply:
Because this uses DefaultConcurrentHashMap, it is _impossible_ for this to be null; so this isn't doing what you want it to do. Look at the implementation of DefaultConcurrentHashMap.